src/gui/properties/peerlistwidget.cpp

```diff
@@ -382,47 +381,64 @@ void PeerListWidget::updatePeer(const BitTorrent::TorrentHandle *torrent, const
+        setModelData(row, PeerListColumns::IP_HIDDEN, peerIp, peerIp);
+
+        itemIter = m_peerItems.insert(peerEndpoint, m_listModel->item(row, PeerListColumns::IP));
         m_itemsByIP[peerEndpoint.address.ip].insert(itemIter.value());
     }
 
     const int row = (*itemIter)->row();
-    m_listModel->setData(m_listModel->index(row, PeerListDelegate::CONNECTION), peer.connectionType());
-    m_listModel->setData(m_listModel->index(row, PeerListDelegate::FLAGS), peer.flags());
-    m_listModel->setData(m_listModel->index(row, PeerListDelegate::FLAGS), peer.flagsDescription(), Qt::ToolTipRole);
-    m_listModel->setData(m_listModel->index(row, PeerListDelegate::CLIENT), peer.client().toHtmlEscaped());
-    m_listModel->setData(m_listModel->index(row, PeerListDelegate::PROGRESS), peer.progress());
-    m_listModel->setData(m_listModel->index(row, PeerListDelegate::DOWN_SPEED), peer.payloadDownSpeed());
-    m_listModel->setData(m_listModel->index(row, PeerListDelegate::UP_SPEED), peer.payloadUpSpeed());
-    m_listModel->setData(m_listModel->index(row, PeerListDelegate::TOT_DOWN), peer.totalDownload());
-    m_listModel->setData(m_listModel->index(row, PeerListDelegate::TOT_UP), peer.totalUpload());
-    m_listModel->setData(m_listModel->index(row, PeerListDelegate::RELEVANCE), peer.relevance());
+
+    setModelData(row, PeerListColumns::CONNECTION, peer.connectionType(), peer.connectionType());
+    setModelData(row, PeerListColumns::FLAGS, peer.flags(), peer.flags(), {}, peer.flagsDescription());
+    const QString client =  peer.client().toHtmlEscaped();
+    setModelData(row, PeerListColumns::CLIENT, client, client);
+    setModelData(row, PeerListColumns::PROGRESS, (Utils::String::fromDouble(peer.progress() * 100, 1) + '%'), peer.progress(), intDataTextAlignment);
+    setModelData(row, PeerListColumns::DOWN_SPEED, Utils::Misc::friendlyUnit(peer.payloadDownSpeed(), 1), peer.payloadDownSpeed(), intDataTextAlignment);
+    setModelData(row, PeerListColumns::UP_SPEED, Utils::Misc::friendlyUnit(peer.payloadUpSpeed(), true), peer.payloadUpSpeed(), intDataTextAlignment);
+    setModelData(row, PeerListColumns::TOT_DOWN, Utils::Misc::friendlyUnit(peer.totalDownload()), peer.totalDownload(), intDataTextAlignment);
+    setModelData(row, PeerListColumns::TOT_UP, Utils::Misc::friendlyUnit(peer.totalUpload()), peer.totalUpload(), intDataTextAlignment);
+    setModelData(row, PeerListColumns::RELEVANCE, (Utils::String::fromDouble(peer.relevance() * 100, 1) + '%'), peer.relevance(), intDataTextAlignment);
 
     const QStringList downloadingFiles {torrent->info().filesForPiece(peer.downloadingPieceIndex())};
-    m_listModel->setData(m_listModel->index(row, PeerListDelegate::DOWNLOADING_PIECE), downloadingFiles.join(';'));
-    m_listModel->setData(m_listModel->index(row, PeerListDelegate::DOWNLOADING_PIECE), downloadingFiles.join('\n'), Qt::ToolTipRole);
+    const QString downloadingFilesDisplayValue = downloadingFiles.join(';');
```

Comment:
Why do you create variable for one "join" result but don't do it for the another one?
P.S. Seems redundant for me. But if it is really useful then why only one of them?

Reply:
this one is required two places while another one is only required once

Reply:
>this one is required at two places while another one is only required once

Sorry, I misread it.